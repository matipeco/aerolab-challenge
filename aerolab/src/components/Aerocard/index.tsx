import { StyledAerocard } from "./style";
import Image from "next/image";

export const Aerocard = () => {
  return (
    <StyledAerocard>
      <div>
        <p className="aerocard-title">Aerocard</p>
        <Image
          src="/assets/icons/aeropay-2.svg"
          alt=""
          width="24"
          height="24"
        />
      </div>
      <div>
        <p>Matias Pecorale</p>
        <p>07/23</p>
      </div>
    </StyledAerocard>
  );
};
