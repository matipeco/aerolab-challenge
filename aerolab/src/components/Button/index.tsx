import { StyledButton } from "./style";
import type { FunctionComponent, PropsWithChildren } from "react";

type Props = PropsWithChildren<{
  className: string;
}>;

export const Button: FunctionComponent<Props> = ({ children, className }) => {
  return <StyledButton className={className}>{children}</StyledButton>;
};
