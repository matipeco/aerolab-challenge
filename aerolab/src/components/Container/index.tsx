import { StyledContainer } from "./style";
import type { FunctionComponent, PropsWithChildren } from "react";

export const Container: FunctionComponent<PropsWithChildren> = ({
  children,
}) => {
  return <StyledContainer>{children}</StyledContainer>;
};
