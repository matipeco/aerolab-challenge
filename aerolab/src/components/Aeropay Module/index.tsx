import { Aerocard } from "../Aerocard";
import { Button } from "../Button";
import { StyledAeropay } from "./style";
import Image from "next/image";
import { RadioButton } from "../RadioButton";

export const AeropayModule = () => {
  return (
    <StyledAeropay>
      <p className="module__title">Add Balance</p>
      <Aerocard />
      <div className="module__points">
        <RadioButton id="point-1000" label="1000" name="points" />
        <RadioButton id="point-5000" label="5000" name="points" />
        <RadioButton id="point-7500" label="7500" name="points" />
      </div>

      <Button className="module__button">
        <Image
          src="/assets/icons/aeropay-3.svg"
          alt=""
          width="24"
          height="24"
        />
        Add Points
      </Button>
    </StyledAeropay>
  );
};
