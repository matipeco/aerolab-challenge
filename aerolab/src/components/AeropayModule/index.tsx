import { Aerocard } from "../Aerocard";
import { Button } from "../Button";
import { StyledAeropay } from "./style";
import Image from "next/image";
import { RadioButton } from "../RadioButton";
import { ChangeEvent, useContext, useState } from "react";
import { Context } from "@/context";

export const AeropayModule = () => {
  const { setPoints } = useContext(Context)!;
  const [selectedPoints, setSelectedPoints] = useState(0);

  const handleChange = (ev: ChangeEvent<HTMLInputElement>) => {
    const { value } = ev.target;
    setSelectedPoints(Number(value));
  };

  const handleClick = () => {
    setPoints((points) => {
      return points + selectedPoints;
    });
  };
  return (
    <StyledAeropay>
      <p className="module__title">Add Balance</p>
      <Aerocard />
      <div className="module__points">
        <RadioButton
          id="point-1000"
          label="1000"
          name="points"
          value="1000"
          onChange={handleChange}
        />
        <RadioButton
          id="point-5000"
          label="5000"
          name="points"
          value="5000"
          onChange={handleChange}
        />
        <RadioButton
          id="point-7500"
          label="7500"
          name="points"
          value="7500"
          onChange={handleChange}
        />
      </div>

      <Button className="module__button" onClick={handleClick}>
        <Image
          src="/assets/icons/aeropay-3.svg"
          alt=""
          width="24"
          height="24"
        />
        Add Points
      </Button>
    </StyledAeropay>
  );
};
