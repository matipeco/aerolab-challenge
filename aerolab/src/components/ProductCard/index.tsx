import { FunctionComponent } from "react";
import { StyledProductCard } from "./style";
import Image from "next/image";
import { Button } from "../Button";

type Props = {
  image: string;
  id: string;
  name: string;
  cost: number;
  category: string;
};

export const ProductCard: FunctionComponent<Props> = ({
  image,
  id,
  name,
  cost,
  category,
}) => {
  return (
    <StyledProductCard>
      <div className="card__container">
        <div className="card__img-container">
          <Image src={image} alt={name} width="280" height="204" />
        </div>
        <div className="card__info">
          <h3 className="card__title">{name}</h3>
          <p className="card__category">{category}</p>
        </div>
      </div>
      <Button className="card__button">
        Redeem for{" "}
        <Image
          src="/assets/icons/aeropay-3.svg"
          alt=""
          width="24"
          height="24"
        />
        {cost}
      </Button>
    </StyledProductCard>
  );
};
