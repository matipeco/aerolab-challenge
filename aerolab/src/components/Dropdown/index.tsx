import { StyledDropdown, StyledContainer } from "./style";
import Image from "next/image";
import { useState, useContext } from "react";
import { AeropayModule } from "../AeropayModule";
import { Context } from "@/context";

export const Dropdown = () => {
  const [isExpanded, setIsExpanded] = useState(false);

  const { points } = useContext(Context)!;

  return (
    <StyledContainer>
      <StyledDropdown onClick={() => setIsExpanded(!isExpanded)}>
        <Image
          src="./assets/icons/aeropay-1.svg"
          alt=""
          width="32"
          height="32"
        />
        <span className="button__points">{points}</span>
        <Image
          src="./assets/icons/chevron-default.svg"
          alt=""
          width="24"
          height="24"
          className={`button__chevron ${isExpanded ? "expanded" : ""}`}
        />
      </StyledDropdown>

      {isExpanded && <AeropayModule />}
    </StyledContainer>
  );
};
