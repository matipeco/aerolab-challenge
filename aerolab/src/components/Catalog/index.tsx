import { Container } from "../Container";
import { StyledCatalog } from "./style";
import type { Product } from "@/pages";
import { ChangeEvent, FunctionComponent, useState } from "react";
import { ProductCard } from "../ProductCard";
import Image from "next/image";
import { RadioButton } from "../RadioButton";

type Props = {
  products: Product[];
};

export const Catalog: FunctionComponent<Props> = ({ products }) => {
  const categories = products.map((prod) => {
    return prod.category;
  });
  const setCategories = Array.from(new Set(categories));

  const [selectedFilter, setSelectedFilter] = useState("");

  const [selectedOrder, setSelectedOrder] = useState("most-recent");

  const catalogProducts = products
    .filter((prod) =>
      selectedFilter ? prod.category === selectedFilter : prod
    )
    .sort((a, b) => {
      if (selectedOrder === "most-recent") {
        return 0;
      }
      return selectedOrder === "highest-price"
        ? b.cost - a.cost
        : a.cost - b.cost;
    });

  const handleChange = (ev: ChangeEvent<HTMLSelectElement>) => {
    setSelectedFilter(ev.target.value);
  };

  const selectOrder = (ev: ChangeEvent<HTMLInputElement>) => {
    setSelectedOrder(ev.target.value);
  };

  return (
    <StyledCatalog>
      <Container>
        <h2 className="catalog__title" id="Catalog">
          <span>tech</span> products
        </h2>
        <div className="filters__container">
          <div className="select__container">
            <label htmlFor="filter">Filter by:</label>
            <select
              name=""
              id="filter"
              className="filters__select"
              onChange={handleChange}
            >
              <option value="">All Products</option>
              {setCategories.map((cat) => {
                return <option key={cat}>{cat}</option>;
              })}
            </select>
          </div>
          <div className="radio__container">
            <p>Sorted by:</p>
            <RadioButton
              id="radio-1"
              name="sort"
              label="Most recent"
              value="most-recent"
              onChange={selectOrder}
              checked={selectedOrder === "most-recent"}
            />
            <RadioButton
              id="radio-2"
              name="sort"
              label="Lowest Price"
              value="lowest-price"
              onChange={selectOrder}
              checked={selectedOrder === "lowest-price"}
            />
            <RadioButton
              id="radio-3"
              name="sort"
              label="Highest Price"
              value="highest-price"
              onChange={selectOrder}
              checked={selectedOrder === "highest-price"}
            />
          </div>

          <div className="pagination__container">
            <button>
              <Image
                src="/assets/icons/chevron-default.svg"
                alt=""
                width="24"
                height="24"
                className="chevron-left"
              />
            </button>

            <p>
              Page <span>1 of 2</span>
            </p>
            <button>
              <Image
                className="chevron-right"
                src="/assets/icons/chevron-default.svg"
                alt=""
                width="24"
                height="24"
              />
            </button>
          </div>
        </div>
        <div className="catalog__products">
          {catalogProducts.map((prod) => (
            <ProductCard
              image={prod.img.url}
              name={prod.name}
              key={prod._id}
              category={prod.category}
              cost={prod.cost}
              id={prod._id}
            />
          ))}
        </div>
        <div className="catalog__footer">
          <p>x of 32 products</p>
          <div className="pagination__container">
            <button>
              <Image
                src="/assets/icons/chevron-default.svg"
                alt=""
                width="24"
                height="24"
                className="chevron-left"
              />
            </button>

            <p>
              Page <span>1 of 2</span>
            </p>
            <button>
              <Image
                className="chevron-right"
                src="/assets/icons/chevron-default.svg"
                alt=""
                width="24"
                height="24"
              />
            </button>
          </div>
        </div>
      </Container>
    </StyledCatalog>
  );
};
