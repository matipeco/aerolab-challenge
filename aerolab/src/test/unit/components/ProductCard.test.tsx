import { ProductCard } from "@/components/ProductCard";
import { render } from "@testing-library/react";

describe("Componente ProductCard", () => {
  test("Se renderiza correctamente", () => {
    const { container } = render(
      <ProductCard
        image="/imagen.png"
        category="phone"
        cost={2300}
        id="id324243"
        name="iphone7"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
