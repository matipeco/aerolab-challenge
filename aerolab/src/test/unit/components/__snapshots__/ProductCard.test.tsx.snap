// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente ProductCard Se renderiza correctamente 1`] = `
<div>
  <article
    class="sc-f637aa53-0 dxiibP"
  >
    <div
      class="card__container"
    >
      <div
        class="card__img-container"
      >
        <img
          alt="iphone7"
          data-nimg="1"
          decoding="async"
          height="204"
          loading="lazy"
          src="/_next/image?url=%2Fimagen.png&w=640&q=75"
          srcset="/_next/image?url=%2Fimagen.png&w=384&q=75 1x, /_next/image?url=%2Fimagen.png&w=640&q=75 2x"
          style="color: transparent;"
          width="280"
        />
      </div>
      <div
        class="card__info"
      >
        <h3
          class="card__title"
        >
          iphone7
        </h3>
        <p
          class="card__category"
        >
          phone
        </p>
      </div>
    </div>
    <button
      class="sc-9ab37ee8-0 hDUWMd card__button  "
    >
      Redeem for
       
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/assets/icons/aeropay-3.svg"
        style="color: transparent;"
        width="24"
      />
      2300
    </button>
  </article>
</div>
`;
